add information on the new Microsoft Test Platform (#1013)
nunit/docs: docs/articles/vs-test-adapter/NUnit-And-Microsoft-Test-Platform.md

# NUnit and Microsoft Test Platform

## Overview

> Microsoft Test Platform (MTP) is the new platform for testing from Microsoft.  The Test Projects can be run as

Reviewer: ```suggestion
Microsoft Test Platform (MTP) is the new platform for testing from Microsoft. The Test Projects can be run as
```

executables, like we do with [NUnitLite](../nunit/running-tests/NUnitLite-Runner.md). There is no longer a test runner;

> the executable **is** the test runner.

Reviewer: ```suggestion
the executable _is_ the test runner.
```

It does take time, however, to develop the same capabilities that we have with NUnit's current system, and there are many historical factors to consider.
But, the NUnit team recognizes that in the meantime, there needs to be a bridge between these two. In NUnit, the NUnit3TestAdapter contains what is
necessary to run the MTP using the same old test runners.

This means we have two modes of MTP operation: either use the new platform but not as an executable, or use it
as an executable too.

> ## Changing a project to use MTP

Reviewer: ```suggestion
## Changing a Project to Use MTP
```

To change a current test project to use MTP, you need to use version 5.0 or greater of the NUnit3TestAdapter.

Note that this version can run both MTP and non-MTP, and you can easily switch between them by setting to project
properties.

In a property group (use the top level one), add the following two properties

```xml
   <EnableNUnitRunner>true</EnableNUnitRunner>
   <OutputType>Exe</OutputType>  
```

The first one enables the MTP.  The second enables it to also run as an executable, but it doesn't prevent you from
using it in Test Explorer or through `dotnet test`.

## Information on the Microsoft Test Platform

* [Microsoft Test Platform](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-intro?tabs=dotnetcli)
* [NUnit Samples](https://github.com/nunit/nunit3-vs-adapter.issues/tree/master/Issue1152)
* [Adapter issue for implementing MTP](https://github.com/nunit/nunit3-vs-adapter/issues/1152)

## Known issues

> In the current version running single tests don't work.  It will always run all tests.

Reviewer: ```suggestion
In the current (v5) version of the NUnit3TestAdapter, running single tests don't work.  It will always run all tests.
```

See [Adapter Issue 1232](https://github.com/nunit/nunit3-vs-adapter/issues/1232).

This applies to both Test Explorer and `dotnet test`.

> As a workaround for the latter, you can filter tests using MTP specific commands

Reviewer: ```suggestion
As a workaround for the latter, you can filter tests using MTP-specific commands
```

Note that this version can run both MTP and non-MTP, and you can easily switch between them by setting to project
properties.

In a property group (use the top level one), add the following two properties

```xml
   <EnableNUnitRunner>true</EnableNUnitRunner>
   <OutputType>Exe</OutputType>  
```

The first one enables the MTP.  The second enables it to also run as an executable, but it doesn't prevent you from
using it in Test Explorer or through `dotnet test`.

## Information on the Microsoft Test Platform

* [Microsoft Test Platform](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-intro?tabs=dotnetcli)
* [NUnit Samples](https://github.com/nunit/nunit3-vs-adapter.issues/tree/master/Issue1152)
* [Adapter issue for implementing MTP](https://github.com/nunit/nunit3-vs-adapter/issues/1152)

## Known issues

In the current version running single tests don't work.  It will always run all tests.
See [Adapter Issue 1232](https://github.com/nunit/nunit3-vs-adapter/issues/1232).

This applies to both Test Explorer and `dotbet test`.

As a workaround for the latter, you can filter tests using MTP specific commands
like `dotnet test -- --filter Name=Test1`.

> 
> 
> 
> 

Reviewer: ```suggestion
```

# NUnit and Microsoft Test Platform

## Overview

Microsoft Test Platform (MTP) is the new platform for testing from Microsoft.  The Test Projects can be run as 

> executables, like we do with [NUnitLite](../nunit/running-tests/NUnitLite-Runner.md). There is no longer a test runner,

Reviewer: ```suggestion
executables, like we do with [NUnitLite](../nunit/running-tests/NUnitLite-Runner.md). There is no longer a test runner;
```

> the executable **is** the test runner. 

Reviewer: ```suggestion
the executable _is_ the test runner. 
```

Author: You need to resuggest, it was overwritten

> It does take time however, to develop the same capabilities that we have with the current system.

Reviewer: ```suggestion
It does take time, however, to develop the same capabilities that we have with NUnit's current system, and there are many historical factors to consider.
```

> In the meantime, there needs to be a bridge between these two. In NUnit, the NUnit3TestAdapter contains what is

Reviewer: ```suggestion
But, the NUnit team recognizes that in the meantime, there needs to be a bridge between these two. In NUnit, the NUnit3TestAdapter contains what is
```

> necessary to run the MTP using the same old test runners. 

Reviewer: ```suggestion
necessary to run the MTP using the same test runners you're currently used to.
```

Author: Again, overwritten

> This mean we have two modes of MTP operation, either just use the new platform, but not as an executable, or use it

Reviewer: ```suggestion
This means we have two modes of MTP operation: either use the new platform but not as an executable, or use it
```

as an executable too.

## Changing a project to use MTP

> To change a current test project to use MTP, you need to use version 5 of the NUnit3TestAdapter.

Reviewer: ```suggestion
To change a current test project to use MTP, you need to use version 5.0 or greater of the NUnit3TestAdapter.
```

Note that this version can run both MTP and non-MTP, and you can easily switch between them by setting to project
properties.

In a property group (use the top level one), add the following two properties

```xml
   <EnableNUnitRunner>true</EnableNUnitRunner>
   <OutputType>Exe</OutputType>  
```

The first one enables the MTP.  The second enables it to also run as an executable, but it doesn't prevent you from
using it in Test Explorer or through `dotnet test`.

> ## Information on the  Microsoft Test Platform

Reviewer: ```suggestion
## Information on the Microsoft Test Platform
```

> [Microsoft Test Platform](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-intro?tabs=dotnetcli)
> 
> [NUnit Samples](https://github.com/nunit/nunit3-vs-adapter.issues/tree/master/Issue1152)
> 
> [Adapter issue for implementing MTP](https://github.com/nunit/nunit3-vs-adapter/issues/1152)

Reviewer: ```suggestion
* [Microsoft Test Platform](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-intro?tabs=dotnetcli)
* [NUnit Samples](https://github.com/nunit/nunit3-vs-adapter.issues/tree/master/Issue1152)
* [Adapter issue for implementing MTP](https://github.com/nunit/nunit3-vs-adapter/issues/1152)
```

## Known issues

> In the current version running single tests don't work.  It will always run all tests.

Reviewer: Which current version, of which aspect of the ecosystem?

Author: Version 5.0.0, which i'm pushing out tomorrow, should be today, but I have been waiting for a fix.

Reviewer: v5 Of the NUnit3TestAdapter I assume you mean?

Author: Also in the beta, you can see it in the issue I've linked to.

Author: Yeah, sorry

Author: The MTP is implemented in the adapter, that is the "bridge" solution.

See [Adapter Issue 1232](https://github.com/nunit/nunit3-vs-adapter/issues/1232).

> This applies to both Test Explorer and `dotbet test`.

Reviewer: ```suggestion
This applies to both Test Explorer and `dotnet test`.
```
